frontend/src/components/pages/CampCreation/CampDetails/index.tsx

```diff
@@ -1,49 +1,520 @@
+  endTime:string | undefined,
+  ageLower:number | undefined,
+  ageUpper:number | undefined,
+  campCapacity:number | undefined,
+  offersEDLP:boolean | undefined,
+  earliestDropOffTime:string | undefined,
+  latestPickUpTime:string | undefined,
+  priceEDLP:number | undefined,
+  addressLine1:string | undefined,
+  addressLine2:string | undefined,
+  city:string | undefined,
+  province:string | undefined,
+  postalCode:string | undefined,
+  campImageURL:string | undefined,
+  handleCampName: (
+    event: React.ChangeEvent<HTMLInputElement>,
+  ) => void;
+  handleCampDescription: (
+    event: React.ChangeEvent<HTMLTextAreaElement>,
+  ) => void;
+  handleDailyCampFee: (
+    event: React.ChangeEvent<HTMLInputElement>,
+  ) => void;
+  handleStartTime: (
+    event: React.ChangeEvent<HTMLInputElement>,
+  ) => void;
+  handleEndTime: (
+    event: React.ChangeEvent<HTMLInputElement>,
+  ) => void;
+  handleAgeLower: (
+    event: React.ChangeEvent<HTMLInputElement>,
+  ) => void;
+  handleAgeUpper: (
+    event: React.ChangeEvent<HTMLInputElement>,
+  ) => void;
+  handleCampCapacity: (
+    event: React.ChangeEvent<HTMLInputElement>,
+  ) => void;
+  toggleEDLP: (
+    event: React.ChangeEvent<HTMLInputElement>,
+  ) => void;
+  handleEarliestDropOffTime: (
+    event: React.ChangeEvent<HTMLInputElement>,
+  ) => void;
+  handleLatestPickUpTime: (
+    event: React.ChangeEvent<HTMLInputElement>,
+  ) => void;
+  handlePriceEDLP: (
+    event: React.ChangeEvent<HTMLInputElement>,
+  ) => void;
+  handleAddressLine1: (
+    event: React.ChangeEvent<HTMLInputElement>,
+  ) => void;
+  handleAddressLine2: (
+    event: React.ChangeEvent<HTMLInputElement>,
+  ) => void;
+  handleCity: (
+    event: React.ChangeEvent<HTMLInputElement>,
+  ) => void;
+  handleProvince: (
+    event: React.ChangeEvent<HTMLSelectElement>,
+  ) => void;
+  handlePostalCode: (
     event: React.ChangeEvent<HTMLInputElement>,
   ) => void;
+  setCampImageURL: React.Dispatch<React.SetStateAction<string | undefined>>
 };
 
 const CampCreationDetails = ({
-  campDetailsDummyOne,
-  campDetailsDummyTwo,
-  campDetailsDummyThree,
-  toggleCampDetailsDummyOne,
-  toggleCampDetailsDummyTwo,
-  handleCampDetailsDummyThree,
-}: CampCreationDetailsProps): React.ReactElement => {
+  campName,
+  campDescription,
+  dailyCampFee,
+  startTime,
+  endTime,
+  ageLower,
+  ageUpper,
+  campCapacity,
+  offersEDLP,
+  earliestDropOffTime,
+  latestPickUpTime,
+  priceEDLP,
+  addressLine1,
+  addressLine2,
+  city,
+  province,
+  postalCode,
+  campImageURL,
+  handleCampName,
+  handleCampDescription,
+  handleDailyCampFee,
+  handleStartTime,
+  handleEndTime,
+  handleAgeLower,
+  handleAgeUpper,
+  handleCampCapacity,
+  toggleEDLP,
+  handleEarliestDropOffTime,
+  handleLatestPickUpTime,
+  handlePriceEDLP,  
+  handleAddressLine1,
+  handleAddressLine2,
+  handleCity,
+  handleProvince,
+  handlePostalCode,
+  setCampImageURL,
+}: CampCreationDetailsProps): JSX.Element => {
```

Comment:
Could we use React.ReactElement here instead? I think JSX.Element is usually used for library components/ is more general purpose if I am not wrong. It handles types a bit differently too IIRC